pandas/core/frame.py

```diff
@@ -4205,8 +4190,13 @@ def set_index(
             return frame
 
     def reset_index(
-        self, level=None, drop=False, inplace=False, col_level=0, col_fill=""
-    ):
+        self,
+        level: Union[Hashable, Tuple, List, None] = None,
+        drop: bool = False,
+        inplace: bool = False,
+        col_level: Hashable = 0,
+        col_fill: Union[Hashable, None] = "",
```

Comment:
```suggestion
        col_fill: Optional[Hashable] = "",
```

We use `Optional[X]` instead of `Union[X, None]`